linux/deploy_vm/deploy_vm_from_iso.yml

```diff
@@ -40,8 +40,13 @@
   include_tasks: ../../common/get_iso_file_list.yml
 
 - name: "Prepare for Ubuntu installation"
-  include_tasks: ubuntu/prepare_ubuntu_iso_install.yml
   when: guest_id is match('ubuntu.*')
+  block:
+    - name: "Check and get upgrade_kernel_version when upgrade_kernel_version is defined"
+      include_tasks: ubuntu/get_mainline_kernel.yml
+      when: upgrade_kernel_version is defined and upgrade_kernel_version
```

Comment:
Please set default value for upgrade_kernel_version to '' at the beginning of this file if it is not defined. Then you can remove all `upgrade_kernel_version is defined` checking, and only check its value in the following tasks.